crates/uv-dispatch/src/lib.rs

```diff
@@ -285,20 +285,15 @@ impl<'a> BuildContext for BuildDispatch<'a> {
             ),
             NoBuild::None => {}
             NoBuild::Packages(packages) => {
+                // We can only prevent builds by name for packages with names. For editable
+                // packages and unnamed requirements, we can't prevent the build.
```

Comment:
Hm.. this is an interesting caveat. We should probably mention this in the `--only-binary` docs?

Reply:
Yeah I don’t really see any way around it. It was also already true for editables, interestingly.

Reply:
We could default to “false” here instead of true. That might be safer? At least for non-editables.

Reply:
That actually seems better since you can always add a package name to fix it.

Reply:
TIL that pip install does not enforce `--no-binary` for direct URL requirements. For example, `pip install "anyio @ https://files.pythonhosted.org/packages/db/4d/3970183622f0330d3c23d9b8a5f52e365e50381fd484d08e3285104333d3/anyio-4.3.0.tar.gz" --only-binary anyio --no-cache --force-reinstall` works fine.